backend/backendutil/search.go

```diff
@@ -104,8 +117,17 @@ func Match(e *message.Entity, seqNum, uid uint32, date time.Time, flags []string
 		}
 	}
 	for _, text := range c.Text {
-		// TODO: also match header fields
-		if ok, err := matchBody(e, text); err != nil || !ok {
+		headerMatch := false
+		for f := e.Header.Fields(); f.Next(); {
+			decoded, err := f.Text()
+			if err != nil {
+				continue
+			}
+			if strings.Contains(f.Key()+": "+decoded, text) {
```

Comment:
I don't remember whether this should match raw fields or decoded fields. Which one is it?

Reply:
RFC 3501 says that fields should be decoded if CHARSET is specified. We do not preserve the information about that argument so the only reasonable action is to decode and hope client specified charset (most probably do).

Reply:
Makes sense!